src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -31,4 +149,53 @@ mdAssemblyRef FindAssemblyRef(
   return mdAssemblyRefNil;
 }
 
+std::vector<Integration> FilterIntegrationsByCaller(
+    const std::vector<Integration>& integrations,
+    const std::wstring& assembly_name) {
+  std::vector<Integration> enabled;
+
+  for (auto& i : integrations) {
+    bool found = false;
+    for (auto& mr : i.method_replacements) {
+      if (mr.caller_method.assembly.name.empty() ||
+          mr.caller_method.assembly.name == assembly_name) {
+        found = true;
+      }
+    }
+    if (found) {
+      enabled.push_back(i);
```

Comment:
should we push the integration into the list and break out of the loop as soon as we find the first match?